src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -426,13 +426,68 @@ internal void AddInterfaceImpl(Type iface)
                 _assembly.EnsureTypeIsVisible(iface);
 
                 _tb.AddInterfaceImplementation(iface);
+
+                // AccessorMethods -> Metadata mappings.
+                var propertyMap = new Dictionary<MethodInfo, PropertyAccessorInfo>();
```

Comment:
Hmm… the other way to avoid the problem might be make AddInterfaceImpl() do three loops – one for properties, one for events, and one for all methods that aren’t associated with a property or event (i.e. just ignore all methods that return true for IsSpecialName)

Then you don’t need all these dictionaries and comparisons to link up the associations.